specifyweb/frontend/js_src/lib/components/FormSliders/IntegratedRecordSelector.tsx

```diff
@@ -173,48 +58,61 @@ export function IntegratedRecordSelector({
 
   const [rawIndex, setIndex] = useSearchParameter(urlParameter);
   const index = f.parseInt(rawIndex) ?? 0;
+
+  const isInteraction =
+    typeof relationship === 'object' &&
+    relationshipIsToMany(relationship) &&
+    typeof collection.related === 'object' &&
+    ['LoanPreparation', 'GiftPreparation', 'DisposalPreparation'].includes(
+      relationship.relatedTable.name
+    );
+
+  const [isDialogOpen, handleOpenDialog, handleCloseDialog] = useBooleanState();
+
   return (
     <ReadOnlyContext.Provider value={isReadOnly}>
-      {formType === 'formTable' ? (
-        <FormTableCollection
-          collection={collection}
-          dialog={dialog}
-          sortField={sortField}
-          viewName={viewName}
-          onAdd={(resources): void => {
-            collection.add(resources);
-            if (typeof handleAdd === 'function') handleAdd(resources);
-          }}
-          onClose={handleClose}
-          onDelete={
-            handleDelete === undefined
-              ? undefined
-              : (_resource, index): void => handleDelete(index, 'minusButton')
-          }
-        />
-      ) : (
-        <RecordSelectorFromCollection
-          collection={collection}
-          defaultIndex={isToOne ? 0 : index}
-          relationship={relationship}
-          onAdd={handleAdd}
-          onDelete={handleDelete}
-          onSlide={(index): void =>
-            typeof urlParameter === 'string'
-              ? setIndex(index.toString())
-              : undefined
+      <RecordSelectorFromCollection
+        collection={collection}
+        defaultIndex={isToOne ? 0 : index}
+        relationship={relationship}
+        isInteraction={isInteraction}
+        onAdd={(resource) => {
+          handleOpenDialog();
```

Comment:
It would be better if you only call this function if `isInteraction`.
Two reasons:
- By doing the check, you are also documenting to anyone reading this code that the interaction dialog should only be open if in the interaction subview.
- While it won't make a difference right now, it might if code is refactored incorrectly - being extra cautious here is not difficult

Reply:
it looks like you applied the fix to a wrong function - I meant you should only call handleOpenDialog() if isInteraction.

though while you are here, the handleAdd call could be simplified: just do `handleAdd?.(resource)l` instead of the `if(...)` check